iroh/src/endpoint.rs

```diff
@@ -992,6 +993,39 @@ impl Endpoint {
         self.msock.direct_addresses()
     }
 
+    /// Returns a [`Watcher`] for any net-reports run from this [`Endpoint`].
+    ///
+    /// A `net-report` checks the network conditions of the [`Endpoint`], such as
+    /// whether it is connected to the internet via Ipv4 and/or Ipv6, its NAT
+    /// status, its latency to the relay servers, and its public addresses.
+    ///
+    /// The [`Endpoint`] continuously runs `net-reports` to monitor if network
+    /// conditions have changed. This [`Watcher`] will return the latest result
+    /// of the `net-report`.
+    ///
+    /// When issuing the first call to this method the first report might
+    /// still be underway, in this case the [`Watcher`] might not be initialized
+    /// with [`Some`] value yet.  Once the net-report has been successfully
+    /// run, the [`Watcher`] will always return [`Some`] report immediately, which
+    /// is the most recently run `net-report`.
+    ///
+    /// # Examples
+    ///
+    /// To get the first report use [`Watcher::initialized`]:
+    /// ```no_run
+    /// use futures_lite::StreamExt;
+    /// use iroh::Endpoint;
+    ///
+    /// # let rt = tokio::runtime::Builder::new_current_thread().enable_all().build().unwrap();
+    /// # rt.block_on(async move {
+    /// let ep = Endpoint::builder().bind().await.unwrap();
+    /// let _report = ep.net_report().initialized().await.unwrap();
+    /// # });
+    /// ```
+    pub fn net_report(&self) -> Watcher<Option<Arc<Report>>> {
```

Comment:
It's a great doc comment.  Though I'd like to make this `#[doc(hidden)]` as I would like to avoid having this in the 1.0 API.  I think it is fairly commonly accepted that doc(hidden) items are not part of a public API.  If we need we could also hide it behind an `experimental_watchable_net_report` cargo feature.

If we hide it maybe the example should be a test instead?  Wait, maybe the example should be turned into a test anyway.  Independent of whether it stays as an example or not.